x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -57,12 +77,22 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
     }
   }, [appTitle, appId, userCanCrud]);
 
+  const value = useMemo(() => {
+    return { ...baseValue, state, dispatch };
+  }, [baseValue, state]);
+
   return isCasesContextValue(value) ? (
-    <CasesContext.Provider value={value}>{children}</CasesContext.Provider>
+    <CasesContext.Provider value={value}>
+      <CasesContextUI state={value.state} />
```

Comment:
I think we can do an optimization here:
Since the `state` coming from the `useReducer` is only used in the `CasesContextUI`, there's no need to put it into the context value. The `dispatch` function is still needed (can be placed directly in the context value initialization). This way the context will never need to be updated by a ui state change, only the CasesContextUI will.
```suggestion
      <CasesContextUI state={state} />
```
Does it make sense?

Reply:
It looks like that's the case but what CasesContextUI does need the context provider. It is the main reason why this code is here.

Reply:
Yes sure `CasesContextUI` needs the context, but that's not what I am saying. I am saying that, unless there will be other components that also need to read this `state` (which I don't think so), there's no need to store the `state` object in the CasesProvider value.

Reply:
ah! my bad I misunderstood that part. Let me try that it makes sense.

Reply:
I made the change, it is all working as expected. Thanks!